src/lib/operators/join_hash.cpp

```diff
@@ -170,15 +163,16 @@ std::shared_ptr<const Table> JoinHash::_on_execute() {
   const auto build_column_type = build_input_table->column_data_type(build_column_id);
   const auto probe_column_type = probe_input_table->column_data_type(probe_column_id);
 
-  // Determine output column order
+  // Depending on which input table became the build/probe table we have to order the columns of the output table.
+  // Semi/Anti* Joins only emit tuples from the probe table
   auto output_column_order = OutputColumnOrder{};
 
   if (_mode == JoinMode::Semi || _mode == JoinMode::AntiNullAsTrue || _mode == JoinMode::AntiNullAsFalse) {
-    output_column_order = OutputColumnOrder::ProbeOnly;
+    output_column_order = OutputColumnOrder::RightOnly;
```

Comment:
Hmm, this will add a little bit of confusion again. We might want to add a few comments here.

@mrks : what's your take on the namings?

Reply:
I remember that we got rid of the entire left/right terminology and probably had a good reason for it. One of it was probably that this becomes confusing once we flip input sides. I presume that this renaming is so that we can share OutputColumnOrder with both join implementations?

Reply:
Yes, but thinking again about it, it might be ok here, since it's only about the output order. You only have to consider the input order for that and that should be quite clear.
Still think we need to add a good comment here, but it should not be the same confusing terminilogy as it was in the hash join internals.

Reply:
Agreed.

Reply:
What about extending the current comment.

Current:
```
// Depending on which input table became the build/probe table we have to order the columns of the output table.
// Semi/Anti* Joins only emit tuples from the probe table
```
New:
```
// Depending on which input table became the build/probe table we have to order the columns of the output table.
// Semi/Anti* Joins only emit tuples from the probe table, which is given by the right input table in Hyrise.
// For other join modes, we need to check which side has been chosen as the probe side (see variable `build_hash_table_for_right_input` for more details).
```

@mrks?

Reply:
Works for me.